packages/node_modules/@webex/plugin-meetings/src/meeting/index.js

```diff
@@ -1021,61 +1032,132 @@ export default class Meeting extends StatelessWebexPlugin {
     this.locusInfo.on(EVENTS.LOCUS_INFO_UPDATE_MEDIA_SHARES, (payload) => {
       const {contentId, disposition} = payload.current;
 
-      // Check if screenshare is being taken over by a remote
-      // If so, terminate the share and allow the remote to be on the floor
-      // Terminating the share allows the user to screenshare again without any issues
       if (
-        this.isSharing &&
-        this.selfId !== contentId &&
-        disposition !== FLOOR_ACTION.RELEASED
+        contentId === payload.previous?.contentId &&
+        disposition === payload.previous?.disposition
       ) {
-        this.updateShare({
-          sendShare: false,
-          receiveShare: this.mediaProperties.mediaDirection.receiveShare
-        }).finally(() => {
-          this.isSharing = false;
+        // nothing changed, so ignore
+        // (this happens when we steal presentation from remote)
+        return;
+      }
 
-          // Trigger remote share start event
-          Trigger.trigger(
-            this,
-            {
-              file: 'meetings/index',
-              function: 'remoteShare'
-            },
-            EVENT_TRIGGERS.MEETING_STARTED_SHARING_REMOTE
-          );
-        });
+      let newShareStatus = this.shareStatus;
+
+      // check if remote started sharing
+      if (
+        this.selfId !== contentId &&
+        disposition === FLOOR_ACTION.GRANTED
+      ) {
+        newShareStatus = SHARE_STATUS.REMOTE_SHARE_ACTIVE;
       }
-      // Check if remote started a share
+      // or if we started sharing
       else if (
-        this.selfId !== contentId &&
+        this.selfId === contentId &&
         disposition === FLOOR_ACTION.GRANTED
       ) {
-        Trigger.trigger(
-          this,
-          {
-            file: 'meetings/index',
-            function: 'remoteShare'
-          },
-          EVENT_TRIGGERS.MEETING_STARTED_SHARING_REMOTE
-        );
+        newShareStatus = SHARE_STATUS.LOCAL_SHARE_ACTIVE;
       }
-      // check if remote stopped a share
+      // or if sharing has been stopped
       else if (
-        this.selfId !== contentId &&
+        payload.previous &&
         disposition === FLOOR_ACTION.RELEASED
       ) {
-        Trigger.trigger(
-          this,
-          {
-            file: 'meetings/index',
-            function: 'remoteShare'
-          },
-          EVENT_TRIGGERS.MEETING_STOPPED_SHARING_REMOTE
-        );
+        newShareStatus = SHARE_STATUS.NO_SHARE;
       }
 
-      this.members.locusMediaSharesUpdate(payload);
+      if (newShareStatus !== this.shareStatus) {
```

Comment:
compare the content ID to check if the remote over takes the share and also pass the content ID as part of the event

Reply:
done.